src/django_components/component.py

```diff
@@ -54,37 +55,113 @@
 RENDERED_COMMENT_TEMPLATE = "<!-- _RENDERED {name} -->"
 
 
-class SimplifiedInterfaceMediaDefiningClass(MediaDefiningClass):
+class ComponentMeta(MediaDefiningClass):
     def __new__(mcs, name: str, bases: Tuple[Type, ...], attrs: Dict[str, Any]) -> Type:
         # NOTE: Skip template/media file resolution when then Component class ITSELF
         # is being created.
         if "__module__" in attrs and attrs["__module__"] == "django_components.component":
             return super().__new__(mcs, name, bases, attrs)
 
         if "Media" in attrs:
-            media: Component.Media = attrs["Media"]
+            media_data: Component.Media = attrs["Media"]
+            _normalize_media(media_data)
+            _normalize_media_filepaths(media_data)
 
-            # Allow: class Media: css = "style.css"
-            if hasattr(media, "css") and isinstance(media.css, str):
-                media.css = [media.css]
+        _resolve_component_relative_files(attrs)
 
-            # Allow: class Media: css = ["style.css"]
-            if hasattr(media, "css") and isinstance(media.css, list):
-                media.css = {"all": media.css}
+        cls = super().__new__(mcs, name, bases, attrs)
 
-            # Allow: class Media: css = {"all": "style.css"}
-            if hasattr(media, "css") and isinstance(media.css, dict):
-                for media_type, path_list in media.css.items():
-                    if isinstance(path_list, str):
-                        media.css[media_type] = [path_list]  # type: ignore
+        # Allow users to provide custom subclasses of Media via `media_class`.
+        # `MediaDefiningClass` defines `media` as a getter (defined in djago.forms.widgets.media_property).
+        # So we reused that and convert it to user-defined Media class
+        media_prop: property = cls.media
+        media_getter = media_prop.fget
+        def media_wrapper(self: "Component") -> Any:
+            if not media_getter:
+                return None
+            media: Media = media_getter(self)
+            return self.media_class(js=media._js, css=media._css)
+        cls.media = property(media_wrapper)
```

Comment:
The bit above makes it possible for users to set `media_class` on the component class to override the Media class. For examples see https://github.com/EmilStenstrom/django-components/pull/526/files#diff-7667a514b72a24c770553fe3421223dcc03252c458518310f25a5ef7746d4669R473 and https://github.com/EmilStenstrom/django-components/pull/526/files#diff-7667a514b72a24c770553fe3421223dcc03252c458518310f25a5ef7746d4669R496